SAMtemplates/lambda_resources.yaml

```diff
@@ -0,0 +1,164 @@
+    AllowedValues:
+      [
+        1,
+        3,
+        5,
+        7,
+        14,
+        30,
+        60,
+        90,
+        120,
+        150,
+        180,
+        365,
+        400,
+        545,
+        731,
+        1096,
+        1827,
+        2192,
+        2557,
+        2922,
+        3288,
+        3653,
+      ]
+
+Conditions:
+  ShouldUseSplunk: !Equals [true, !Ref EnableSplunk]
+  AllowAccessToSpine: !Equals [true, !Ref AccessToSpine]
+
+Resources:
+  LambdaRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: lambda.amazonaws.com
+            Action:
+            - sts:AssumeRole
+      ManagedPolicyArns:
+        - !ImportValue lambda-resources:LambdaInsightsLogGroupPolicy
+        - !ImportValue account-resources:LambdaEncryptCloudwatchKMSPolicy
+        - !ImportValue account-resources:LambdaDecryptSecretsKMSPolicy
+        - !Ref DynamoDBReadPolicy
+        - !Ref DynamoDBWritePolicy
+        - !If
+          - AllowAccessToSpine
+          - !ImportValue account-resources:LambdaAccessSecretsPolicy
+          - !Ref AWS::NoValue
+
+  LambdaManagedPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      Roles:
+        - !Ref LambdaRole
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Action:
+              - logs:CreateLogStream
+              - logs:PutLogEvents
+            Resource:
+              - !Sub "arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:${LambdaLogGroup}"
+              - !Sub "arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:${LambdaLogGroup}:log-stream:*"
+
+  DynamoDBReadPolicy:
+      Type: AWS::IAM::ManagedPolicy
+      Properties:
+        PolicyDocument:
+          Version: "2012-10-17"
+          Statement:
+            - Effect: Allow
+              Action:
+                - dynamodb:GetItem
+                - dynamodb:Scan
+                - dynamodb:Query
+                - dynamodb:BatchGetItem
+                - dynamodb:DescribeTable
+                - "dynamodb:GetRecords"
+                - "dynamodb:ListTables"
+              Resource:
+                - !Sub "arn:${AWS::Partition}:dynamodb:${AWS::Region}:${AWS::AccountId}:table/${TableName}"
+                - !Sub "arn:${AWS::Partition}:dynamodb:${AWS::Region}:${AWS::AccountId}:table/${TableName}/index/*"
+
+  DynamoDBWritePolicy:
+      Type: AWS::IAM::ManagedPolicy
+      Properties:
+        PolicyDocument:
+          Version: "2012-10-17"
+          Statement:
+            - Effect: Allow
+              Action:
+                - "dynamodb:BatchWriteItem"
+                - "dynamodb:DeleteItem"
+                - "dynamodb:PutItem"
+                - "dynamodb:UpdateItem"
+                - "dynamodb:CreateTable"
+                - "dynamodb:UpdateTable"
```

Comment:
CreateTable & UpdateTable should probably be removed from the list, as these policies should just be operating on items in the designated table.

Reply:
Unnecessary actions have been removed.